File: Pythonleetcode/Leetcode343.py
class Solution(object):
    def integerBreak(self, n):
        if n==2:
            return 1
        elif n==3:
            return 2
        else:
            ans=1
            while(n!=0):
                if n==4:
                    return ans*4
                if n==2:
                    return ans*2
                if n==3:
                    return ans*3
                else: n=n-3
                ans=ans*3





print(Solution().integerBreak(2))
print(Solution().integerBreak(10))

File: CppPractice/Project1/Project1/Leetcode343.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int integerBreak(int n) {
        vector<int> dp(n + 1, 1);
        int temp = 0;
        for (int i = 2; i <= n; i++)
        {
            for (int j =1; j < i; j++)
            {
                temp = max(temp,j*dp[i-j]);
                temp = max(temp,j*(i-j));
            }
            dp[i] = temp;
            temp = 0;
        }
        return dp[n];

    }
};
